agent/lib/kontena/workers/log_worker.rb

```diff
@@ -199,14 +204,10 @@ def on_container_event(topic, event)
       error exc
     end
 
-    # Close the log queue, stopping processing once empty. It cannot be restarted anymore.
-    def close
-      @queue.close
-    end
-
     def finalize
       stop_streaming if streaming?
-      close
+      sleep 1 until @queue.empty?
```

Comment:
This doesn't really work, you can't sleep in the finalizer task... it will just raise `TaskTerminated` as part of the `cleanup` immediately following.

Reply:
Should be fixed now.